dart/collision/CollisionGroup.cpp

```diff
@@ -0,0 +1,209 @@
+ *   MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ *   DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+ *   CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ *   SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ *   LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
+ *   USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ *   AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *   LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *   ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *   POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "dart/collision/CollisionGroup.h"
+
+#include <cassert>
+
+#include "dart/collision/CollisionObject.h"
+#include "dart/collision/CollisionDetector.h"
+#include "dart/dynamics/BodyNode.h"
+#include "dart/dynamics/Skeleton.h"
+
+namespace dart {
+namespace collision {
+
+//==============================================================================
+CollisionGroup::CollisionGroup(const CollisionDetectorPtr& collisionDetector)
+  : mCollisionDetector(collisionDetector)
+{
+  assert(mCollisionDetector);
+}
+
+//==============================================================================
+CollisionGroup::~CollisionGroup()
+{
+  // Do nothing
+}
+
+//==============================================================================
+CollisionDetector* CollisionGroup::getCollisionDetector()
+{
+  return mCollisionDetector.get();
+}
+
+//==============================================================================
+const CollisionDetector* CollisionGroup::getCollisionDetector() const
+{
+  return mCollisionDetector.get();
+}
+
+//==============================================================================
+void CollisionGroup::addShapeFrame(const dynamics::ShapeFrame* shapeFrame)
+{
+  if (!shapeFrame)
+    return;
+
+  if (hasShapeFrame(shapeFrame))
+    return;
+
+  auto collObj = mCollisionDetector->claimCollisionObject(shapeFrame);
+
+  notifyCollisionObjectAdded(collObj.get());
+
+  mShapeFrames.push_back(shapeFrame);
+  mCollisionObjects.push_back(collObj);
+}
+
+//==============================================================================
+void CollisionGroup::addShapeFrames(
+    const std::vector<const dynamics::ShapeFrame*>& shapeFrames)
+{
+  for (const auto& shapeFrame : shapeFrames)
+    addShapeFrame(shapeFrame);
+}
+
+//==============================================================================
+void CollisionGroup::addShapeFramesOf()
+{
+  // Do nothing
+}
+
+//==============================================================================
+void CollisionGroup::removeShapeFrame(const dynamics::ShapeFrame* shapeFrame)
+{
+  if (!shapeFrame)
+    return;
+
+  const auto search
+      = std::find_if(mShapeFrames.begin(), mShapeFrames.end(),
+                     [&](const dynamics::ShapeFrame* it)
+                     { return it == shapeFrame; });
```

Comment:
`shapeFrame` exactly matches the `value_type` of `mShapeFrames`, so I am not sure why `std::find_if` is necessary here. Shouldn't `std::find(mShapeFrames.begin(), mShapeFrames.end(), shapeFrame)` work?